osu.Game.Rulesets.Catch/UI/CatchPlayfield.cs

```diff
@@ -45,14 +44,14 @@ public CatchPlayfield(BeatmapDifficulty difficulty)
         [BackgroundDependencyLoader]
         private void load()
         {
-            var trailContainer = new Container
+            var trailDisplay = new CatcherTrailDisplay
             {
                 Anchor = Anchor.BottomLeft,
                 Origin = Anchor.TopLeft
             };
             var droppedObjectContainer = new DroppedObjectContainer();
 
-            Catcher = new Catcher(trailContainer, droppedObjectContainer, difficulty)
+            Catcher = new Catcher(trailDisplay, droppedObjectContainer, difficulty)
```

Comment:
At this point could the catcher create the pieces on its own internally and then expose them as public so they can be proxied into the hierarchy in the desired order? I think it's a bit weird that it has to receive *two* drawables from above now.

Reply:
I agree this would be better if the `CatcherTrailDisplay` was constructed internally, exposed publicly and then proxied below.

Reply:
An issue is that the `CatcherTrailDisplay` shouldn't be moved when the catcher moves.

Reply:
...which may not have been an issue if `CatcherArea` still existed...

I guess they could be created internally but not added in catcher, and the parent would add it on its own instead of proxying? I dunno, that'd be turbo weird.

Or just move some smaller portion of `Catcher`. Honestly this feels like going in circles over and over again.

Reply:
`CatcherArea` exists, and the question where to put the trail generation logic.
The previous situation was the logic was scattered in both `CatcherTrailDisplay` and `Catcher`, and I thought it is suboptimal.
This PR proposes the logic to be moved to `Catcher`.
Another way is move the logic to `CatcherArea`, which makes some sense (as evidenced by the position relativity issue).

Reply:
Oh, my bad, I thought it got nuked entirely, but it was only that the catcher that was moved out of it.

Yeah I dunno. I'd have to go and try the alternative, see what issues fall out, and compare the end result.